src/arch/riscv/common_riscv.lds

```diff
@@ -29,7 +29,7 @@ SECTIONS
     {
         *(.boot.data)
     }
-    . = ALIGN(64K);
```

Comment:
Once the kernel has finished booting, all of the memory from KERNEL_ELF_BASE to ki_boot_end is turned into untypeds and released to userlevel.  This alignment seems like it was trying to ensure that at least a 64K untyped could be created.  I can't find any earlier history than 2014 which would shed any light about why it was introduced in the Arm linker scripts. 4K untyped is useful enough and any rounding up here that pushes the rest of the kernel image up is just going to steal memory that would be turned into untypeds at the other end.

Reply:
I am assuming that the 64K alignment is to support userspace code that uses 64K frame objects; for some combinations of boot size and kernel size, a 64K aligned kernel might need 1 frame for boot + 2 for runtime, while a 4K aligned kernel frees the same amount of memory _but_ in two pieces that cannot be assembled into a 64K frame.

This is moot on riscv since the smallest architectural large frame is 2M.